### PR TITLE
[flutter_image] Adopt code excerpts in README

### PR DESCRIPTION
**Pull request by mike-v2**
Updates the README to use a compiled excerpt source for its example of using `NetworkImageWithRetry`.

Part of [flutter/flutter#102679](https://github.com/flutter/flutter/issues/102679)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/flutter_image/example/test/readme_excerpts_test.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import 'package:flutter_image/flutter_image.dart';
+import 'package:flutter_image_example/readme_excerpts.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('networkImageWithRetry returns an Image with NetworkImageWithRetry', () {
+    // Ensure that the snippet code runs successfully.
+    final Image result = networkImageWithRetry();
+
+    // It should have a image property of the right type
```
Comment (stuartmorgan) [inline packages/flutter_image/example/test/readme_excerpts_test.dart:15]: Nit: missing period.

File: packages/flutter_image/example/test/readme_excerpts_test.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    expect(result.image, isInstanceOf<NetworkImageWithRetry>());
+    // And the NetworkImageWithRetry should have a url property
```
Comment (stuartmorgan) [inline packages/flutter_image/example/test/readme_excerpts_test.dart:17]: Same.

**Review by stuartmorgan — APPROVED** [review]
LGTM with comment nits. Thanks!

(This is a case where we could potentially rework the actual example code some and then excerpt from that, instead of having a special-purpose file, but I don't think it's worth the churn of adding another image to the example.)

**Comment by stuartmorgan** [comment]
@tarrinneal for secondary review.

**Review by tarrinneal — APPROVED** [review]

**Pull request merged**